dockstore-integration-testing/src/test/java/io/dockstore/client/cli/CheckUrlHelperFullIT.java

```diff
@@ -104,14 +110,67 @@ void settingVersionMetadata() throws Exception {
         assertTrue(workflowVersion.getVersionMetadata().isPublicAccessibleTestParameterFile(), "Should be public because the descriptor has no parameters at all");
     }
 
+    /**
+     * Tests the updateOpenData endpoint. It's not easy to simulate the LambdaUrlChecker -- the
+     * Dockstore config file just takes the lambda checker url. So this test currently
+     * gets around that by:
+     * <ul>
+     *     <li>Workflows that have no file inputs are open</li>
+     *     <li>Workflows that file inputs and no test parameter files are closed</li>
+     *     <li>Invalid, e.g., missing the primary descriptor, are unknown</li>
+     * </ul>
+     *
+     * TODO: Run the lambda checker with AWS SAM, mock the responses, or make the lambda checker
+     * injectable via the dockstore.yml (there's already an interface).
+     */
+    @Test
+    void testUpdateOpenData() {
+        CommonTestUtilities.cleanStatePrivate2(EXT.getTestSupport(), false, testingPostgres);
+        ApiClient client = getOpenAPIWebClient(USER_2_USERNAME, testingPostgres);
+        io.dockstore.openapi.client.api.EntriesApi
+            entriesApi = new io.dockstore.openapi.client.api.EntriesApi(client);
+        WorkflowsApi workflowsApi = new WorkflowsApi(client);
+        assertEquals(0, rowsWithPublicAccessibleData());
+        Workflow wdlWorkflow = workflowsApi.manualRegister(SourceControl.GITHUB.name(), "DockstoreTestUser2/hello-dockstore-workflow", "/Dockstore.wdl", "",
+            DescriptorLanguage.WDL.getShortName(), "");
+        wdlWorkflow = workflowsApi.refresh1(wdlWorkflow.getId(), true);
+        Workflow cwlWorkflow = workflowsApi.manualRegister(SourceControl.GITHUB.name(), "DockstoreTestUser2/hello-dockstore-workflow", "/Dockstore.cwl", "cwlworkflow",
+            DescriptorLanguage.CWL.getShortName(), "");
+        cwlWorkflow = workflowsApi.refresh1(cwlWorkflow.getId(), true);
+
+        // Clear publicaccessibletestparameterfile
+        testingPostgres.runUpdateStatement("update version_metadata set publicaccessibletestparameterfile = null");
+        // Confirm the above worked cleared out the language versions
+        wdlWorkflow = workflowsApi.getWorkflow(wdlWorkflow.getId(), null);
+        wdlWorkflow.getWorkflowVersions().forEach(wv -> assertNull(wv.getVersionMetadata().isPublicAccessibleTestParameterFile()));
+        cwlWorkflow = workflowsApi.getWorkflow(cwlWorkflow.getId(), null);
+        cwlWorkflow.getWorkflowVersions().forEach(wv -> assertNull(wv.getVersionMetadata().isPublicAccessibleTestParameterFile()));
+
+        final Integer processed = entriesApi.updateOpenData(Boolean.TRUE);
+        assertEquals(2, processed);
+
+        wdlWorkflow = workflowsApi.getWorkflow(wdlWorkflow.getId(), null);
+        wdlWorkflow.getWorkflowVersions().forEach(wv -> assertTrue(!wv.isValid() || wv.getVersionMetadata().isPublicAccessibleTestParameterFile(),
+            "No valid versions have file inputs, should be open"));
```

Comment:
> If I understand correctly, this is using `DockstoreTestUser2/hello-dockstore-workflow` Will this eventually break if someone adds a proper test parameter file in the future?

Yes, it would break. Your comment also makes me realize that the way I'm testing for specific version counts is fragile, e.g., if somebody adds a new tag, the count assertions would break. I'll update.